crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -904,3 +907,196 @@ fn given_hardcoded_contract_set_erc20_balance_to_zero() {
         );
     });
 }
+
+#[test]
+fn given_hardcoded_contract_run_invoke_tx_v3_fails_sender_not_deployed() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        let none_origin = RuntimeOrigin::none();
+
+        // Wrong address (not deployed)
+        let contract_address = ContractAddress(PatriciaKey(
+            StarkFelt::try_from("0x03e437FB56Bb213f5708Fcd6966502070e276c093ec271aA33433b89E21fd32f").unwrap(),
+        ));
+
+        let mut transaction = get_invoke_v3_dummy(Starknet::chain_id(), NONCE_ZERO);
+        if let starknet_api::transaction::InvokeTransaction::V3(tx) = &mut transaction.tx {
+            tx.sender_address = contract_address;
+        };
+
+        assert_err!(Starknet::invoke(none_origin, transaction), Error::<MockRuntime>::AccountNotDeployed);
+    })
+}
+
+#[test]
+fn given_hardcoded_contract_run_invoke_tx_v3_on_argent_account_then_it_works() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+        let none_origin = RuntimeOrigin::none();
+        // NOT WORKING
+        let chain_id = Starknet::chain_id();
+        let mut transaction = get_invoke_v3_argent_dummy(chain_id);
+        if let starknet_api::transaction::InvokeTransaction::V3(tx) = &mut transaction.tx {
+            tx.signature = sign_message_hash(transaction.tx_hash);
+        };
+
+        assert_ok!(Starknet::invoke(none_origin, transaction));
+    });
+}
+
+#[test]
+fn given_hardcoded_contract_run_invoke_tx_v3_on_openzeppelin_account_then_it_works() {
```

Comment:
I don't think you have to write test for all the different wallets.
The wallets tests don't change. So everything is already tested by the other tests

Reply:
What we want you to test is what is constitutive of the txV3, which is the new fee mechanism using STRK

Reply:
Ok, i get it. I'm going to erase the tests related to wallets. Where can i find that new fee mechanism to see if i can understand it and write some tests to them?

Reply:
> In the code I'm seeing some tests related to strk and eth fees, i have to do something like that? Or is already tested?


I don't think so. Just write one test for each txV3, checking it works correctly

Reply:
> > In the code I'm seeing some tests related to strk and eth fees, i have to do something like that? Or is already tested?
> 
> I don't think so. Just write one test for each txV3, checking it works correctly

I know that i have to do the tests in `invoke_ts.rs` file. But i dont get exactly where is the functionality that i have to test, to see if i can understand it and do something.

Reply:
What do you mean by saying "one test for each txV3"? Like in the `mod.rs` file?

Reply:
invokeV3, DeclareV3, deploy_accountV3
Those are the 3 tx types